homeassistant/components/media_player/liveboxplaytv.py

```diff
@@ -0,0 +1,209 @@
+"""
+Support for interface with an Orange Livebox Play TV appliance.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/media_player.liveboxplaytv/
+"""
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+
+import homeassistant.util as util
+from homeassistant.components.media_player import (
+    SUPPORT_TURN_ON, SUPPORT_TURN_OFF, SUPPORT_PLAY,
+    SUPPORT_NEXT_TRACK, SUPPORT_PAUSE, SUPPORT_PREVIOUS_TRACK,
+    SUPPORT_VOLUME_STEP, SUPPORT_VOLUME_MUTE, SUPPORT_SELECT_SOURCE,
+    MEDIA_TYPE_CHANNEL, MediaPlayerDevice, PLATFORM_SCHEMA)
+from homeassistant.const import (
+    CONF_HOST, CONF_PORT, STATE_ON, STATE_OFF, STATE_PLAYING,
+    STATE_PAUSED, STATE_UNKNOWN, CONF_NAME)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['liveboxplaytv==1.4.1']
+
+_CONFIGURING = {}  # type: Dict[str, str]
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'Livebox Play TV'
+DEFAULT_PORT = 8080
+
+SUPPORT_LIVEBOXPLAYTV = SUPPORT_TURN_OFF | SUPPORT_TURN_ON | \
+    SUPPORT_NEXT_TRACK | SUPPORT_PAUSE | SUPPORT_PREVIOUS_TRACK | \
+    SUPPORT_VOLUME_STEP | SUPPORT_VOLUME_MUTE | SUPPORT_SELECT_SOURCE | \
+    SUPPORT_PLAY
+
+MIN_TIME_BETWEEN_SCANS = timedelta(seconds=10)
+MIN_TIME_BETWEEN_FORCED_SCANS = timedelta(seconds=1)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string
+})
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Orange Livebox Play TV platform."""
+    host = config.get(CONF_HOST)
+    port = config.get(CONF_PORT)
+    name = config.get(CONF_NAME)
+
+    if host is None:
```

Comment:
This can't happen as far as I can see. `vol.Required(CONF_HOST)` is set, thus `host` will never be `None`.